drivers/input/input_kbd_matrix.c

```diff
@@ -293,14 +312,16 @@ static void input_kbd_matrix_polling_thread(void *arg1, void *unused2, void *unu
 	ARG_UNUSED(unused3);
 
 	while (true) {
-		input_kbd_matrix_drive_column(dev, INPUT_KBD_MATRIX_COLUMN_DRIVE_ALL);
-		api->set_detect_mode(dev, true);
+		if (!input_kbd_matrix_is_suspended(dev)) {
+			input_kbd_matrix_drive_column(dev, INPUT_KBD_MATRIX_COLUMN_DRIVE_ALL);
+			api->set_detect_mode(dev, true);
 
-		/* Check the rows again after enabling the interrupt to catch
-		 * any potential press since the last read.
-		 */
-		if (api->read_row(dev) != 0) {
-			input_kbd_matrix_poll_start(dev);
+			/* Check the rows again after enabling the interrupt to catch
+			 * any potential press since the last read.
+			 */
+			if (api->read_row(dev) != 0) {
+				input_kbd_matrix_poll_start(dev);
+			}
```

Comment:
Applying this PR to the it82xx2_evb for testing, we observed that the power number remains continuously active. The reason for the power number staying active seems to stem from the state you mentioned in the fourth point: after reading the actual rows value and emitting keys and then doing another dummy round, it should stop at line 327`(k_sem_take(&data->poll_lock, K_FOREVER);`). However, upon tracing the code, the semaphore count is 1 here, preventing the `input_kbd_matrix_polling_thread `from entering suspend. Should we consider resetting the semaphore's count to zero after confirming there are no rows to read?
As follows:
```
if (api->read_row(dev) != 0) {
	input_kbd_matrix_poll_start(dev);
} else {
        k_sem_reset(&data->poll_lock);
}
```

Reply:
But it the semaphore should only be 1 once, then it should fall back and lock. Hey could it be that what you are observing is the PM subsystem resuming the device with the CPU? Could you try with `CONFIG_PM_DEVICE_SYSTEM_MANAGED=n`? I bumped into this problem as well and I'm trying to figure a plan, probably using runtime PM for the keyboard.

Reply:
> Hey could it be that what you are observing is the PM subsystem resuming the device with the CPU?

Yes.

After setting CONFIG_PM_DEVICE_SYSTEM_MANAGED=n, the keyboard will not remain in the active state.

Reply:
Ok, thanks for confirming that, I opened https://github.com/zephyrproject-rtos/zephyr/issues/75963 to try and figure out the story, for now I'm afraid the only option is to use `DEVICE_PM_RUNTIME` so that system management leaves this one alone, could you test that combination too? (`CONFIG_PM_DEVICE_SYSTEM_MANAGED=y` and `CONFIG_DEVICE_PM_RUNTIME=y`)

Reply:
When both configurations are enabled, pressing a key does not trigger an interrupt. According to the debug messages, it seems that after the first boot, the device was set to `PM_DEVICE_ACTION_SUSPEND `and never executed `input_kbd_matrix_pm_action()` again, resulting in the detection mode not being set.

Reply:
Sorry, yes if you enable PM_RUNTIME then the device starts in suspend mode, you have to sneak in a `pm_device_runtime_get` to activate it somewhere, something like:

```c
#include <zephyr/pm/device_runtime.h>
const struct device *dev =  DEVICE_DT_GET(DT_NODELABEL(kbd));
pm_device_runtime_get(dev);
```

Reply:
Yes, if I add `pm_device_runtime_get `before `pm_device_action_run`, the keyboard will operate normally, thanks.

Reply:
ok, thanks for for testing this, so this should be good to go but we need to get to the bottom of the PM story, see the other issue